scripts/dts/gen_defines.py

```diff
@@ -918,8 +926,14 @@ def write_global_macros(edt):
     out_comment('Bus information for status "okay" nodes of each compatible\n')
     for compat, buses in compat2buses.items():
         for bus in buses:
-            out_define(
-                f"DT_COMPAT_{str2ident(compat)}_BUS_{str2ident(bus)}", 1)
+            if isinstance(bus, list):
+                bus_list = bus
+            else:
+                bus_list = [ bus ]
+
+            for one_bus in bus_list:
+                out_define(
+                    f"DT_COMPAT_{str2ident(compat)}_BUS_{str2ident(one_bus)}", 1)
```

Comment:
This is too clunky and error-prone. People are going to forget to check the list type, and forcing this boilerplate on every use site is not good API. I would prefer an approach where we add a new property, `buses`, which always returns a list, replacing `bus`.

Reply:
Updated the Node class to have 'buses' and 'on-buses' so they are both lists. For Binding node, added 'buses' to return a list of bus(es), though there is no 'on-buses' as this is supposed to be a single string.

Reply:
> Updated the Node class to have 'buses' and 'on-buses' so they are both lists. 

So the above if/else block should be simplified now that the type is known, right?